Add stats bar to transparency page

moteanup24 opened the pull request:
Build preview:
- [/transparency](https://os-website-git-fork-moteanup24-statistics-bar-opensats.vercel.app/transparency)

moteanup24 commented:
@dergigi let me know what standard procedure is for handling the .env variables wrt PRs since I only have them on my local machine. Thanks!

dergigi commented:
@moteanup24 our staging environments have their own `.env` definitions, so we'll have to add the two variables that you added to these environments. And for production too, eventually.

dergigi commented:
I got it working locally, so well done! 

I'd suggest adding the component to `MDXComponents.tsx` so we can use it any `.mdx` file, including the one that renders the /transparency page.

Also, make sure that the component renders nicely in dark mode and on mobile. Our use of tailwind should make that relatively straightforward.

---

<details>

<summary>Right now it looks like this, which isn't ideal:</summary>

![Screenshot 2025-01-28 at 16 52 00](https://github.com/user-attachments/assets/db36388d-4c2c-4f1d-a3a4-b3ceed0ec93c)

</details>

dergigi commented on pages/transparency.tsx:
```diff
@@ -2,22 +2,94 @@ import { InferGetStaticPropsType } from 'next'
 
 const DEFAULT_LAYOUT = 'PageLayout'
+const GOOGLE_DOC_ID = process.env.NEXT_PUBLIC_GOOGLE_DOC_ID
+const GOOGLE_API_KEY = process.env.NEXT_PUBLIC_GOOGLE_API_KEY
```
It would be better to use proper credentials (something like `client_id`, `client_secret`, and a `refresh_token`) as opposed to a simple `API_KEY` that is unscoped.

moteanup24 replied:
@dergigi to be sure, api keys in this context are read-only. 

See: https://theoephraim.github.io/node-google-spreadsheet/#/guides/authentication ("API key access does not need any scopes selected - as it is read only")

But, assuming there are other advantages to the other authentication methods that I haven't yet grasped, do you have a preference for service account vs Oauth 2.0?

dergigi commented on pages/transparency.tsx:
```diff
@@ -2,22 +2,94 @@ import { InferGetStaticPropsType } from 'next'
+        }
+        setStatistics(stats)
+      } catch (e) {
+        console.log('Error!', e)
```
Something more descriptive than "Error!" would be good 😅

dergigi commented on pages/transparency.tsx:
```diff
@@ -2,22 +2,94 @@ import { InferGetStaticPropsType } from 'next'
-      MDXComponents={MDXComponents}
-    />
+    <>
+      <StatsBar />
```
If you make `StatsBar` an `MDXComponent` we can render it inside the LayoutRenderer, which gives us more options (i.e. we aren't forced to display it at the very top).

dergigi commented:
Again: well done! 🙌  Some comments and change requests, but with a little bit of refactoring and polishing this will be good to go soon!

moteanup24 commented:
Pushed a few updates:
- refactored StatsBar as an MDX Component
- fixed dark mode; I hard coded the colors based on my own visual inspection but pls let me know if I should reference dark mode settings defined elsewhere

Now working on other options for the authentication mode but happy to take comments/requests in the meantime!

dergigi commented:
Awesome! These are great updates 💪 

I just noticed an edge case: when the screen is a certain size the numbers will blend into each other, like this:
<img width="711" alt="Screenshot 2025-01-31 at 23 56 03" src="https://github.com/user-attachments/assets/fdaf97f7-0ffa-4413-82e9-821527dd645a" />

What we did in the past for issues like this is format the large numbers as ~22.1M or $100k etc. See [`formatUsd`](https://github.com/OpenSats/website/blob/16b4d62bb55808457428c57192b154a84dbe5493/pages/funds/%5B...slug%5D.tsx#L48) for details. (We will have to extend this to deal with billions for "sats sent" 😅)

moteanup24 commented:
Authentication type update still in the works!

dergigi commented:
> Authentication type update still in the works!

Fantastic! Let me know in case you need any help 🙏

dergigi commented:
Before I forget: You'll probably have to rebase with `master` (or merge the `master` changes into your branch) and run `npm i` again to resolve the `package-lock.json` conflicts.

dergigi commented:
Different PR was opened:

- https://github.com/OpenSats/website/pull/406
